### PR TITLE
[DO NOT MERGE] withObservables, without Rx

### PR DESCRIPTION
**Pull request by radex**
This is hand-written asynchronous logic to completely remove Rx dependency from withObservables. It certainly makes the code uglier, harder to read, and more error-prone… but this is low-level infra, we have hundreds or thousands of these, and there's a measurable performance impact of the old implementation.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -64,6 +84,8 @@ function getTriggeringProps<PropsInput: {}>(
   return propNames.map(name => props[name])
 }
 
+const hasOwn = Object.prototype.hasOwnProperty
```
Comment (kokusGr) [inline src/index.js:87]: Any particular reason you assigned this to a variable instead of using it directly?

**Review by kokusGr — APPROVED** [review]
LGTM 👍  Error handling has been merged, so is there anything else that's blocking the merge?

File: src/index.js (hunk 1 above)
Comment (radex) [reply src/index.js:87]: not really, I don't remember why i did that ;)

**Comment by radex** [comment]
@kokusGr Nope, feel free to merge

**Comment by kokusGr** [comment]
Ok, merging. Thanks :)

**Pull request merged**